app/assets/javascripts/discourse/widgets/header.js.es6

```diff
@@ -340,6 +350,36 @@ export default createWidget('header', {
         return Ember.get(ctx, 'type');
       }
     }
+  },
+
+  applyAutocomplete($searchInput) {
+    const siteSettings = this.siteSettings;
+
+    $searchInput.autocomplete({
+      template: findRawTemplate('category-tag-autocomplete'),
+      key: '#',
+      width: '100%',
+      treatAsTextarea: true,
+      transformComplete(obj) {
```

Comment:
I think we can try to extract the logic here into a function so that we don't have to repeat the same logic twice.